### DISCUSSION TITLE
[Vue] Add `null` as a possible value for `Listbox` and `ListboxOption` in the type definitions

### DISCUSSION THREAD

**Discussion by kerunix** [tailwindlabs/headlessui#2026] [Ideas]
Hello there and thank you so much for the great job you're doing at Tailwind Labs, you guys are great !

I'm working on a select menu in a Vue 3 app using `Listbox` as my base component. My select menu needs to accept `null` as a possible value. Technically it works at runtime with no problem but Volar is not happy about the type declarations of Listbox.

```vue
<!-- inputValue is nullable -->
 <Listbox
    :model-value="inputValue" <!-- Type 'null' is not assignable to type 'string | number | boolean | Record<string, any> | undefined'. -->
  >
     <!-- code -->
</Listbox>
```

The current type declarations are:
```
modelValue: { type: [Object, String, Number, Boolean], default: undefined } // For the  Listbox component
value: { type: [Object, String, Number, Boolean] } // For the ListboxOption component
```

Do you see any problem with adding `null` as a possible type ?

I'm willing to work on a PR for this if needed although I'm not sure how that would go for the React package since I don't know anything about React.

**Comment by laurensiusadi** [comment]
Any workaround before the package support `null` value?
